packages/table/src/TableHead.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import HeaderRow from './HeaderRow';
+import TableHeader from './TableHeader';
+import { TableProps } from './Table';
+import { isComponentType } from '@leafygreen-ui/lib';
+import { Types, useTableContext } from './TableContext';
+
+type TableHeaderProps = Pick<TableProps, 'data' | 'columns'>;
+
+const TableHead = ({ columns = [], data }: TableHeaderProps) => {
+  const {
+    state: { sort },
+    dispatch,
+  } = useTableContext();
+  const usingHeaderRow = React.useRef(false);
+
+  const sortRows = (columnId: number, key: string) => {
+    dispatch({
+      type: Types.SortTableData,
+      payload: {
+        columnId,
+        key,
+        data,
+      },
+    });
+  };
+
+  function createCols(array: Array<React.ReactNode>): React.ReactNode {
+    return array.map((child, index) => {
+      if (isComponentType(child, 'HeaderRow')) {
+        usingHeaderRow.current = true;
+
+        const { children } = child?.props;
+
+        return React.cloneElement(child, {
+          children: createCols(React.Children.toArray(children)),
+        });
+      }
+
+      if (isComponentType(child, 'TableHeader')) {
+        const { label, accessor: accessorProp } = child.props;
+
+        let glyph = 'Unsorted';
+
+        const accessor = accessorProp || label?.toLowerCase();
```

Comment:
This doesn't handle calling accessor if it's a function.

What we did in mms is normalized the data such that the accessor is always a function. What we also did is when the accessor was a function is we passed in each datum object when sorting so you could do custom value look ups.

For example if you have a complex array like:

```jsx
const data = [
    {
        name: 'harry',
        job: {
            team: 'cloud',
            title: 'punmaster'
        },
    }
];

// Accessor could be:
<TableHeader
  accessor={(data) => data.job.title}
>
```

For convienience we also let you do dot lookup as a string, so with the above example you could have done `accessor="job.title"` but I don't think anyone really knows that exist.

Luckily we have an [existing implementation of this normalize behavior in mms.](https://github.com/10gen/mms/blob/624053ad04198e130771ebd0d723e1dd6c7011f3/client/js/common/components/Table/TablePlain.js#L16-L33)

Reply:
Got it! Just need to add tests for this, but I believe we have this working as intended.